<?xml version="1.0"?>

<launch>
    <arg name="initial_x" default="0.0"/>
    <arg name="initial_y" default="0.0"/>
    <group ns="robot1">
        <!-- <include file="$(find realsense2_camera)/launch/rs_camera.launch">
            <arg name="tf_prefix" value="robot1_camera"/>
        </include> -->
        <node name="interfacer" pkg="robomaster_interface" type="pub_data.py">
            <param name="robot_number" value="robot1"/>
        </node>

        <node pkg="robot_pose_ekf" type="robot_pose_ekf" name="robot_pose_ekf">
            <param name="output_frame" value="robot1_odom_combined"/>
            <param name="base_footprint_frame" value="robot1_base_footprint"/>
            <param name="freq" value="30.0"/>
            <param name="sensor_timeout" value="1.0"/>
            <param name="odom_used" value="true"/>
            <param name="imu_used" value="true"/>
            <param name="vo_used" value="false"/>
        </node>
        <node pkg="tf" type="static_transform_publisher" name="imu_broadcaster" args="0 0 0 0 0 0 robot1_base_link robot1_imu 30"/>
        <node pkg="tf" type="static_transform_publisher" name="base_link_broadcaster" args="0 0 0 0 0 0 robot1_base_footprint robot1_base_link 30"/>
        <node pkg="tf" type="static_transform_publisher" name="camera_base_broadcaster" args="0.095 0.05 0.2 0 0 0 robot1_base_footprint robot1_camera_link 30" />
        <node name="handler" pkg="robomaster_interface" type="robot_handler.py">
            <param name="robot_number" value="robot1"/>
            <param name="initial_x" value="$(arg initial_x)"/>
            <param name="initial_y" value="$(arg initial_y)"/>
            <param name="confidence_level" value="0.9"/>
            <param name="magnitude_limit" value="0.2"/>
            <param name="safety_radius" value="0.5"/>
        </node>
        <node name="camera_handler" pkg="robomaster_interface" type="camera_handler.py">
            <param name="robot_number" value="robot1"/>
        </node>     
        <node name="lidar_handler" pkg="robomaster_interface" type="lidar_handler.py">
            <param name="robot_number" value="robot1"/>
            <param name="angle_window" value="5"/>
        </node>
        
    </group>

</launch>